SofaKernel/modules/SofaCore/src/sofa/core/objectmodel/BaseData.cpp

```diff
@@ -161,24 +163,53 @@ void BaseData::update()
     {
         (*it)->updateIfDirty();
     }
+
+    /// Check if there is a parent (so a predecessor in the DDG), if so
+    /// update the internal value.
     auto parent = parentData.resolvePathAndGetTarget();
     if (parent)
     {
 #ifdef SOFA_DDG_TRACE
         if (m_owner)
             dmsg_warning(m_owner) << "Data " << m_name << ": update from parent " << parentBaseData->m_name;
 #endif
-        updateFromParentValue(parent);
+        updateValueFromLink(parent);
         // If the value is dirty clean it
         if(this->isDirty())
         {
             cleanDirty();
         }
     }
+    _doOnUpdate_();
+}
+
+bool BaseData::updateValueFromLink(const BaseData* parent)
+{
+    /// Try if the fast path succeeded, in general this means that the two Data are
+    /// of exactly the same internal types.
+    if(_doSetValueFromLink_(parent))
+    {
+        return true;
+    }
+    return copyValueFrom(parent);
 }
 
 /// Update this Data from the value of its parent
-bool BaseData::updateFromParentValue(const BaseData* parent)
+bool BaseData::copyValueFrom(const BaseData* parent)
+{
+    /// Try if the fast path succeeded, in general this means that the two Data are
+    /// of exactly the same internal types.
+    if(_doCopyValueFrom_(parent))
+    {
+        return true;
+    }
+    /// If the fast path didn't succeeded we try to do copy using a much less efficient approach.
+    return genericCopyValueFrom(parent);
```

Comment:
I'm not sure I would add this call to try to resolve a failing copy. Do you think there will be a lot of case were the normal copy (using dynamic_cast) will failed and this will solve it?
For me a copyValueFrom is a "short and secure" operation, if it fails, this means something is wrong in your code/way of using it.

Reply:
This is how sofa was working so I preserve it.

Reply:
oh ok.. didn't see...